nodeup/pkg/model/kubelet_test.go

```diff
@@ -245,9 +269,26 @@ func RunGoldenTest(t *testing.T, basedir string, key string, builder func(*Nodeu
 	nodeupModelContext, err := BuildNodeupModelContext(basedir)
 	if err != nil {
 		t.Fatalf("error loading model %q: %v", basedir, err)
-		return
 	}
-	nodeupModelContext.KeyStore = &fakeKeyStore{T: t}
+
+	keystore := &fakeCAStore{}
+	keystore.T = t
+	keystore.privateKeys = map[string]*pki.PrivateKey{
+		"ca":                      mustParsePrivateKey(dummy_key),
+		"apiserver-aggregator-ca": mustParsePrivateKey(dummy_key),
+		"kube-controller-manager": mustParsePrivateKey(dummy_key),
+		"kube-proxy":              mustParsePrivateKey(dummy_key),
+		"kube-scheduler":          mustParsePrivateKey(dummy_key),
```

Comment:
Would be slightly better to use different certs/keys to catch cases where the code used the wrong CA, but then it's not likely a human would catch the wrong key when reviewing the expected output.

Reply:
I don't disagree; I debated actually removing the keys entirely before comparison, but I figured that would be more confusing.

It's the flags and other pod settings (e.g. resources) that I think are particularly valuable in these tests.

Reply:
I suppose I'd need to access `pki.IssueCert` through an interface that the tests would mock out.

Reply:
Ah - I have two ideas on that front then. 

* If you've put cert generation into a task (which isn't a bad idea, you might be doing that already) then I think we'll be testing the right thing - the task invocation arguments, not the filesystem write calls.

* Otherwise we can post-process the serialized yaml and either remove the keys or replace them with their key characteristics (e.g. "CN=foo, Validity=Client")

Are you planning on putting cert generation into a task?

Reply:
#9172

Reply:
I'll take a look at #9130, to get closer to #9172.  Your approach looks good.  I don't know exactly how the TaskDependentResource will get serialized, but I imagine we can make it work nicely!

We can always defer this until #9172 has landed ... we've gotten by without it so far!